sql_generators/active_users_aggregates_v4/__init__.py

```diff
@@ -0,0 +1,198 @@
+        },
+        {
+            "name": "nightly",
+            "table": "`moz-fx-data-shared-prod.org_mozilla_focus_nightly_live.baseline_v1`",
+        },
+    ],
+    "klar_ios": [
+        {"table": "`moz-fx-data-shared-prod.org_mozilla_ios_klar_live.baseline_v1`"}
+    ],
+    "klar_android" : [
+        {"table" : "`moz-fx-data-shared-prod.org_mozilla_klar_live.baseline_v1`"}
+    ],
+}
+
+class Browsers(Enum):
+    """Enumeration with browser names and equivalent dataset names."""
+
+    firefox_desktop = "Firefox Desktop"
+    fenix = "Fenix"
+    focus_ios = "Focus iOS"
+    focus_android = "Focus Android"
+    firefox_ios = "Firefox iOS"
+    klar_ios = "Klar iOS"
+    klar_android = "Klar Android"
+
+
+@click.command()
+@click.option(
+    "--output-dir",
+    "--output_dir",
+    help="Output directory generated SQL is written to",
+    type=click.Path(file_okay=False),
+    default="sql",
+)
+@click.option(
+    "--target-project",
+    "--target_project",
+    help="Google Cloud project ID",
+    default="moz-fx-data-shared-prod",
+)
+@use_cloud_function_option
+def generate(target_project, output_dir, use_cloud_function):
+    """Generate per-app queries, views and metadata for active users and search counts aggregates.
+
+    The parent folders will be created if not existing and existing files will be overwritten.
+    """
+    env = Environment(loader=FileSystemLoader(str(THIS_PATH / "templates")))
+    output_dir = Path(output_dir) / target_project
+    # query templates
+    mobile_query_template = env.get_template("mobile_query.sql")
+    desktop_query_template = env.get_template("desktop_query.sql")
+    focus_android_query_template = env.get_template("focus_android_query.sql")
+    # view templates
+    focus_android_view_template = env.get_template("focus_android_view.sql")
+    mobile_view_template = env.get_template("mobile_view.sql")
+    view_template = env.get_template("view.sql")
+    # metadata template
+    metadata_template = "metadata.yaml"
+    # schema template
+    desktop_schema_template = "desktop_schema.yaml"
+    mobile_schema_template = "mobile_schema.yaml"
+    # checks templates
+    desktop_checks_template = env.get_template("desktop_checks.sql")
+    fenix_checks_template = env.get_template("fenix_checks.sql")
+    mobile_checks_template = env.get_template("mobile_checks.sql")
+
+    for browser in Browsers:
+        if browser.name == "firefox_desktop":
+            query_sql = reformat(
+                desktop_query_template.render(
+                    app_value=browser.value,
+                )
+            )
+            schema_template = desktop_schema_template
+        elif browser.name == "focus_android":
+            query_sql = reformat(
+                focus_android_query_template.render(
+                    project_id=target_project,
+                    app_name=browser.name,
+                )
+            )
+            schema_template = mobile_schema_template
```

Comment:
Why does focus_android have a query template that's distinct from the mobile query template?

Reply:
It was the only browser that required keeping legacy and Glean data, probably something to reevaluate soon.